client/app/components/permissions-editor/PermissionsEditorDialog.jsx

```diff
@@ -0,0 +1,183 @@
+
+  const debouncedSearchUsers = useCallback(debounce(
+    search => searchUsers(search)
+      .then(setUsers)
+      .finally(() => setLoadingUsers(false)),
+    DEBOUNCE_SEARCH_DURATION,
+  ), []);
+
+  useEffect(() => {
+    setLoadingUsers(true);
+    debouncedSearchUsers(searchTerm);
+  }, [searchTerm]);
+
+  return (
+    <Select
+      className="w-100 m-b-10"
+      placeholder="Add users..."
+      showSearch
+      onSearch={setSearchTerm}
+      suffixIcon={loadingUsers ? <i className="fa fa-spinner fa-pulse" /> : <i className="fa fa-search" />}
+      filterOption={false}
+      notFoundContent={null}
+      value={undefined}
+      getPopupContainer={trigger => trigger.parentNode}
+      onSelect={onSelect}
+    >
+      {users.map(user => (
+        <Option key={user.id} value={user.id} disabled={isUserDisabled(user)}>
+          <UserPreviewCard user={user}>
+            {previewCardAddon(user)}
+          </UserPreviewCard>
+        </Option>
+      ))}
+    </Select>
+  );
+}
+
+UserSelect.propTypes = {
+  onSelect: PropTypes.func,
+  previewCardAddon: PropTypes.func,
+  isUserDisabled: PropTypes.func,
+};
+UserSelect.defaultProps = { onSelect: () => {}, previewCardAddon: () => null, isUserDisabled: () => false };
+
+function PermissionsEditorDialog({ dialog, owner, context, aclUrl }) {
+  const [loadingGrantees, setLoadingGrantees] = useState(true);
+  const [grantees, setGrantees] = useState([]);
+  const { loadGrantees, addPermission, removePermission } = useGrantees(aclUrl);
+  const loadUsersWithPermissions = useCallback(() => {
+    setLoadingGrantees(true);
+    loadGrantees()
+      .then(setGrantees)
+      .catch(() => notification.error('Failed to load grantees list'))
+      .finally(() => setLoadingGrantees(false));
+  }, []);
+
+  const userHasPermission = useCallback(
+    user => (user.id === owner.id || !!get(find(grantees, { id: user.id }), 'accessType')),
+    [grantees],
+  );
+
+  useEffect(() => {
+    loadUsersWithPermissions();
+  }, [aclUrl]);
+
+  return (
+    <Modal
+      {...dialog.props}
+      className="permissions-editor-dialog"
+      title={<PermissionsEditorDialogHeader context={context} />}
+      footer={(<Button onClick={dialog.dismiss}>Close</Button>)}
+    >
+      <UserSelect
+        onSelect={userId => addPermission(userId).then(loadUsersWithPermissions)}
+        previewCardAddon={user => (userHasPermission(user) ? '(already has permission)' : null)}
+        isUserDisabled={user => userHasPermission(user)}
+      />
+      <h5>Users with permissions</h5>
+      {!loadingGrantees ? (
+        <div className="scrollbox p-5 m-t-10" style={{ maxHeight: '40vh' }}>
+          <List
+            size="small"
+            dataSource={[owner, ...grantees]}
+            renderItem={user => (
+              <List.Item>
+                <UserPreviewCard key={user.id} user={user}>
+                  {user.id === owner.id ? (<Tag>Owner</Tag>) : (
```

Comment:
I'd rename to `author`

Reply:
Also, reset the `.ant-tag` right margin so it aligns to the edge.

Reply:
> I'd rename to `author`

Author does look better, but do you think it can be confusing in code, considering all the other places use `owner`? (including the backend)

Reply:
I do that stuff all the time and i think it’s ok as long as it says author only on the presentation layer.

Reply:
Currently there is no difference  between Owner and Author. But it's very possible we will allow changing the user reference here and in this case semantically it becomes more correct to call it Owner, no?

Reply:
It's not that important. Just making sure users understand the label and I think `author` is clearer.